src/odemis/gui/comp/canvas.py

```diff
@@ -945,15 +951,39 @@ def _draw_merged_images(self, ctx, interpolate_data=False):
                 else:
                     md = im.metadata
 
-                if md['blend_mode'] == BLEND_SCREEN:
+                blend_mode = md['blend_mode']
+                if n == 1: # For single image, don't use merge ratio
                     merge_ratio = 1.0
-                elif i == n - 1: # last image
-                    if n == 1:
-                        merge_ratio = 1.0
-                    else:
-                        merge_ratio = self.merge_ratio
                 else:
-                    merge_ratio = 1 - i / n
+                    # If there are all "screen" (= last one is screen):
+                    # merge ratio   im0   im1
+                    #     0         1      0
+                    #    0.25       1      0.5
+                    #    0.5        1      1
+                    #    0.75       0.5    1
+                    #     1         0      1
```

Comment:
Is this correct?

Reply:
Yes, at least, this is what the code does, and this gives a "intuitive" behaviour. That's a little odd, that's why I've described it in a long comment ;-)